onecodex/models/sample.py

```diff
@@ -25,6 +25,40 @@ def where(cls, *filters, **keyword_filters):
         instances_route = 'instances' if not public else 'instances_public'
         limit = keyword_filters.get('limit', None if not public else 1000)
 
+        # handle conversion of tag UUIDs or names to Tags objects
+        tags = keyword_filters.pop('tags', [])
+
+        if not isinstance(tags, list):
+            tags = [tags]
+
+        new_tags = []
+
+        for t in tags:
+            if isinstance(t, Tags):
+                new_tags.append(t)
+            else:
+                # is it a uuid?
+                if len(t) == 16:
```

Comment:
Let's write an `is_ocx_id` helper function that does this plus and alphabet check. Something like:

```
assert set(x for x in name).issubset(string.hexdigits)
```